src/main/scala/it/pagopa/interop/agreementmanagement/model/persistence/AgreementPersistentBehavior.scala

```diff
@@ -41,6 +37,25 @@ object AgreementPersistentBehavior {
 
         agreement.fold(handleFailure(_)(replyTo), persistStateAndReply(_, AgreementAdded)(replyTo))
 
+      case UpdateAgreement(updated, replyTo) =>
+        val agreement: Either[Throwable, PersistentAgreement] = state.agreements
+          .get(updated.id.toString)
+          .map(_ => updated)
+          .toRight(AgreementNotFound(updated.id.toString))
+
+        agreement.fold(handleFailure(_)(replyTo), persistStateAndReply(_, AgreementUpdated)(replyTo))
+
+      case DeleteAgreement(agreementId, replyTo) =>
+        val agreement: Option[PersistentAgreement] = state.agreements.get(agreementId)
+
+        agreement
+          .map(_ =>
+            Effect
+              .persist(AgreementDeleted(agreementId))
+              .thenRun((_: State) => replyTo ! StatusReply.Success(()))
```

Comment:
There's the API `.thenReply(replyTo)((_: State) => message)` too if you want

Reply:
I left the situation unchanged as we are using `StatusReply` in the return types.
I would postpone this possible change to a future task that also involves the other persistence services.

Reply:
That's fine, but I want to clarify that `message` could even be a `StatusReply.Success(_)`.